Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/NativeMethods.cs

```diff
@@ -0,0 +1,191 @@
+        //
+        public const int PERF_MULTI_COUNTER = 0x02000000;  // sum of multiple instances
+
+        //
+        //  Select one of the following values to indicate the display suffix (if any)
+        //
+        public const int PERF_DISPLAY_NOSHOW = 0x40000000;  // value is not displayed
+
+        //
+        //  Predefined counter types
+        //
+
+        // A label: no data is associated with this counter (it has 0 length)
+        // Do not display.
+        public const int PERF_COUNTER_NODATA =
+                (PERF_SIZE_ZERO | PERF_DISPLAY_NOSHOW);
+        
+        // The divisor for a sample, used with the previous counter to form a
+        // sampled %.  You must check for >0 before dividing by this!  This
+        // counter will directly follow the  numerator counter.  It should not
+        // be displayed to the user.
+        public const int PERF_SAMPLE_BASE =
+                (PERF_SIZE_DWORD | PERF_TYPE_COUNTER | PERF_COUNTER_BASE |
+                PERF_DISPLAY_NOSHOW |
+                0x00000001);  // for compatibility with pre-beta versions
+
+       // Used as the denominator In the computation of time or count
+        // averages.  Must directly follow the numerator counter.  Not dis-
+        // played to the user.
+        public const int PERF_AVERAGE_BASE =
+                (PERF_SIZE_DWORD | PERF_TYPE_COUNTER | PERF_COUNTER_BASE |
+                PERF_DISPLAY_NOSHOW |
+                0x00000002);  // for compatibility with pre-beta versions
+        
+        // Number of instances to which the preceding _MULTI_..._INV counter
+        // applies.  Used as a factor to get the percentage.
+        public const int PERF_COUNTER_MULTI_BASE =
+                (PERF_SIZE_LARGE | PERF_TYPE_COUNTER | PERF_COUNTER_BASE |
+                PERF_MULTI_COUNTER | PERF_DISPLAY_NOSHOW);
+        
+        // Indicates the data is a base for the preceding counter which should
+        // not be time averaged on display (such as free space over total space.)
+        public const int PERF_RAW_BASE =
+                (PERF_SIZE_DWORD | PERF_TYPE_COUNTER | PERF_COUNTER_BASE |
+                PERF_DISPLAY_NOSHOW |
+                0x00000003);  // for compatibility with pre-beta versions
+
+        public const int PERF_LARGE_RAW_BASE =
+                    (PERF_SIZE_LARGE | PERF_TYPE_COUNTER | PERF_COUNTER_BASE |
+                    PERF_DISPLAY_NOSHOW);
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal class PERF_COUNTER_DEFINITION
+        {
+            public int ByteLength = 0;
+            public int CounterNameTitleIndex = 0;
+
+            // this one is kind of weird. It is defined as in SDK:
+            // #ifdef _WIN64
+            //  DWORD           CounterNameTitle;
+            // #else
+            //  LPWSTR          CounterNameTitle;
+            // #endif
+            // so we can't use IntPtr here.
+
+            public int CounterNameTitlePtr = 0;
```

Comment:
Relax, this mapping will never change :) It would be a bit expensive to remove native types in the entire codebase... We currently have 309 occurences of "int" alone in the Web SDK :)